interfacer/src/browsh/vim_mode.go

```diff
@@ -0,0 +1,594 @@
+package browsh
+
+import (
+	"encoding/json"
+	"reflect"
+	"strings"
+	"time"
+	"unicode"
+
+	"github.com/atotto/clipboard"
+	"github.com/gdamore/tcell"
+)
+
+// TODO: A little description as to the respective responsibilties of this code versus the
```

Comment:
@tobimensch could you answer the TODO briefly?

Reply:
Mode based keyboard navigation

Reply:
I disagree about the word "only". Getting the link hint positions is the hardest part and therefore we're lucky we can use vimium code for it and don't have to event the wheel. And I disagree about it for a second reason, because vimium.js contains a lot of useful code besides link hinting, that I strongly believe we're going to use. For example I'm using vimium.js also for going to the next/previous page on websites like Google, which requires finding the correct links in the page, but is separate from the link hinting code. So vimium.js is an integral part of the whole keyboard navigation.
You're right that we receive the link positions in the go code and display it directly in the terminal, contrary to how it's done in the original vimium. I think this gives us the best of both worlds.

Reply:
Yeah, I've got a better idea of what's going on now. It's definitely more than just link hinting.